### PR TITLE
[IMP] open_academy: Add access rules to different kind of users, wizard to save new attendees on sessions and spanish translation  T#52898

### PR DESCRIPTION
**Pull request by rolandojduartem**
Add access to different kind of users (admin, manager and user) in the session and course model.
Screenshot from 2022-01-19 16-16-17
Screenshot from 2022-01-19 16-12-31

Also, I added a recor rule to prevent the managers from updating records if there is a responsible assigned.
Screenshot from 2022-01-20 09-31-57

### CODE REVIEW

**Comment by rolandojduartem** [comment]
I added a wizard to let the user can update sessions with new attendees
Screenshot from 2022-01-21 14-11-47

Also, I added the spanish translation.
Screenshot from 2022-01-21 14-12-27

**Comment by rolandojduartem** [comment]
@frahikLV, In this pull request, I finished security, wizard and internationalization sections from Odoo course. Could you review it please?

File: open_academy/data/ir_rule.xml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+<odoo>
+    <record id="restrict_write_unlink" model="ir.rule">
+        <field name="name">There is a responsible already!!</field>
```
Comment (frahikLV) [inline open_academy/data/ir_rule.xml:3]: This is not a message, it's the name of the record
```suggestion
        <field name="name">Open Academy: Write or delete own courses</field>
```

File: open_academy/demo/res_users.xml [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    <record id="res_partner_john_smith" model="res.partner">
+        <field name="name">John Smith(USER)</field>
+        <field name="company_name">YourCompany</field>
+        <field name="street">250 Executive Park Blvd, Suite 3400</field>
+        <field name="city">San Francisco</field>
+        <field name="zip">94134</field>
+        <field name="phone">+1 (650) 555-0111 </field>
+        <field name="email">info@yourcompany.example.com</field>
+        <field name="website">www.example.com</field>
+    </record>
```
Comment (frahikLV) [inline open_academy/demo/res_users.xml:11]: Move this to res.partner file

File: open_academy/demo/res_users.xml [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    <record id="res_partner_king_mcdonald" model="res.partner">
+        <field name="name">King McDonald(MANAGER)</field>
+        <field name="company_name">YourCompany</field>
+        <field name="street">250 Executive Park Blvd, Suite 3400</field>
+        <field name="city">San Francisco</field>
+        <field name="zip">94134</field>
+        <field name="phone">+1 (650) 555-0111 </field>
+        <field name="email">info@yourcompany.example.com</field>
+        <field name="website">www.example.com</field>
+    </record>
```
Comment (frahikLV) [inline open_academy/demo/res_users.xml:27]: move this to res.partner file

File: open_academy/wizards/wizard_save_attendee.py [hunk 4]
```diff
@@ -0,0 +1,15 @@
+from odoo import models, fields, Command
+
+
+class Wizard(models.TransientModel):
+    _name = 'wizard_save_attendee'
+    _description = 'Save attendees'
+    session_ids = fields.Many2many(
+        'session',
+        required=True,
+        default=lambda self: self.env['session'].browse(self._context.get('active_id'))
```
Comment (frahikLV) [inline open_academy/wizards/wizard_save_attendee.py:10]: Remove this, please send the ids on the action call the ids via context with `default_session_ids` key

File: open_academy/wizards/wizard_save_attendee.py [hunk 5]
```diff
@@ -0,0 +1,15 @@
+from odoo import models, fields, Command
+
+
+class Wizard(models.TransientModel):
+    _name = 'wizard_save_attendee'
```
Comment (frahikLV) [inline open_academy/wizards/wizard_save_attendee.py:5]: change all the references of the name of the wizard from `wizard_save_attendee` to `assign.attendee.sessions`
modes `_name` don't use an underscore in the name, uses `.` , change this when you rename the wizard

File: open_academy/wizards/wizard_save_attendee.py [hunk 6]
```diff
@@ -0,0 +1,15 @@
+from odoo import models, fields, Command
+
+
+class Wizard(models.TransientModel):
```
Comment (frahikLV) [inline open_academy/wizards/wizard_save_attendee.py:4]:
```suggestion
class AssignAttendeeSessions(models.TransientModel):
```

File: open_academy/wizards/wizard_save_attendee.py [hunk 7]
```diff
@@ -0,0 +1,15 @@
...
+    _description = 'Save attendees'
+    session_ids = fields.Many2many(
+        'session',
+        required=True,
+        default=lambda self: self.env['session'].browse(self._context.get('active_id'))
+    )
+    attendee_ids = fields.Many2many('res.partner')
+
+    def save_several_session(self):
+        self.session_ids.write({'attendee_ids': [Command.link(attendee.id) for attendee in self.attendee_ids]})
```
Comment (frahikLV) [inline open_academy/wizards/wizard_save_attendee.py:15]: Add a validation that cannot assign more attendees than available seats

File: open_academy/wizards/wizard_save_attendee.xml [hunk 8]
```diff
@@ -0,0 +1,29 @@
+<odoo>
+    <record id="wizard_save_attendee_view_form" model="ir.ui.view">
+        <field name="name">wizard_save_attendee.view.form</field>
+        <field name="model">wizard_save_attendee</field>
+        <field name="arch" type="xml">
+            <form string="">
```
Comment (frahikLV) [inline open_academy/wizards/wizard_save_attendee.xml:6]:
```suggestion
            <form>
```

File: open_academy/wizards/wizard_save_attendee.py [hunk 9]
```diff
@@ -0,0 +1,15 @@
...
+    _name = 'wizard_save_attendee'
+    _description = 'Save attendees'
+    session_ids = fields.Many2many(
+        'session',
+        required=True,
+        default=lambda self: self.env['session'].browse(self._context.get('active_id'))
+    )
+    attendee_ids = fields.Many2many('res.partner')
+
+    def save_several_session(self):
```
Comment (frahikLV) [inline open_academy/wizards/wizard_save_attendee.py:14]:
```suggestion
    def action_register_attendees_to_sessions(self):
```

**Review by frahikLV — CHANGES_REQUESTED** [review]
@rolandojduartem can you address these changes?

File: open_academy/data/ir_rule.xml [hunk 10]
```diff
@@ -0,0 +1,12 @@
...
+        <field name="model_id" ref="model_course"/>
+        <field name="groups" eval="[Command.link(ref('group_open_academy_manager'))]"/>
```
Comment (frahikLV) [inline open_academy/data/ir_rule.xml:5]:
```suggestion
        <field name="groups" ref="group_open_academy_manager"/>
```

**Review by frahikLV — CHANGES_REQUESTED** [review]
@rolandojduartem also fix this

**Comment by frahikLV** [comment]
Rename your PR and all the long commits with a short message... Say that doesn't import long messages, but, not too long...

File: open_academy/data/ir_rule.xml (hunk 10 above)
Comment (rolandojduartem) [reply open_academy/data/ir_rule.xml:5]: The original was fine. Done!

**Comment by rolandojduartem** [comment]
@frahikLV changes were applied, I do not know if you are going to review it or Germana will do it.

**Comment by rolandojduartem** [comment]
Hi, @goliveirab Francisco had left some commentaries and I tried to address the changes, could you review it, please?

**Comment by goliveirab** [comment]
LGTM...

**Pull request merged**